core/src/test/java/nl/nn/adapterframework/util/CounterStatisticTest.java

```diff
@@ -0,0 +1,67 @@
+package nl.nn.adapterframework.util;
+
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * CounterStatistic Tester.
+ *
+ * @author <Sina Sen>
+
+ */
+public class CounterStatisticTest {
+
+
+    /**
+     *
+     * Method: performAction(int action)
+     *
+     */
+    @Test
+    public void testPerformActionSummaryOrFull() throws Exception {
+        CounterStatistic cs = new CounterStatistic(10);
+        cs.performAction(0);
+        assertEquals(cs.getValue(), 10);
+    }
+
+    /**
+     *
+     * Method: performAction(int action)
+     *
+     */
+    @Test
+    public void testPerformActionReset() throws Exception {
+        CounterStatistic cs = new CounterStatistic(10);
+        cs.performAction(2);
+        assertEquals(cs.getValue(), 0);
+    }
+    
+
+    /**
+     *
+     * Method: performAction(int action)
+     *
+     */
+    @Test
+    public void testPerformActionMarkMainOrFull() throws Exception {
+        CounterStatistic cs = new CounterStatistic(10);
+        cs.performAction(3);
+        assertEquals(cs.getValue(), 10);
+    }
```

Comment:
You are testing the actions here, instead of using the integer values, 123, could you use the more descriptive `HasStatistics` integer fields? So action 2 would become `HasStatistics.STATISTICS_ACTION_RESET`.

Maybe you could also write 2 more tests for action 3 and 4 (mark main and mark full)

Reply:
will add.

Reply:
added